tasks/src/main/scala/dagr/tasks/ScatterGather.scala

```diff
@@ -53,9 +54,41 @@ object ScatterGather {
     def sources: Option[Seq[Source]]
   }
 
+  /** Wraps a set of results as a [[Partitioner]] that produces those results as partitions. */
+  private class FixedPartitioner[Result](results: Iterable[Result]) extends SimpleInJvmTask with Partitioner[Result] {
+    var partitions: Option[Seq[Result]] = None
+    override def run(): Unit = this.partitions = Some(results.toSeq)
```

Comment:
I would suggest using `results.toIndexedSeq` to make sure you get a strict and not lazily evaluated seq from results.